src/sorcha/modules/PPLinkingFilter.py

```diff
@@ -49,105 +48,50 @@ def PPLinkingFilter(
-        # as there aren't enough tracklets to count as a full track
-        if len(obs_above_min_sep["night"].unique()) < min_tracklets:
-            continue
-
-        unique_nights = obs_above_min_sep["night"].unique()
-
-        # checks to see if a track containing a number of tracklets >= min_tracklets
-        # exists in a window of days <= tracklet_interval.
-        # assumes all observations in a night are part of a valid tracklet
-        # which should be true at this point
-
-        # once a linking is made, the night on which the linking was made is stored
-        linked_night = False
-        for i, night in enumerate(unique_nights):
-            if i + min_tracklets - 1 >= len(unique_nights):
-                break
-
-            diff = unique_nights[i + min_tracklets - 1] - night
-
-            if diff < tracklet_interval:
-                linked_night = night
-                break
-
-        # get all observations of this object made on or after the night it was linked
-        if linked_night:
-            obs_final_night = obs_object[obs_object["night"] >= linked_night]
+    # create a small dataframe for difi to work on with only the relevant columns
+
+    difi_dataframe = pd.DataFrame(
+        {
+            "object_id": observations["ObjID"].astype(object),
+            "obs_id": observations["FieldID"],
+            "time": observations["FieldMJD"],
+            "night": observations["night"],
+            "ra": observations["AstRA(deg)"],
+            "dec": observations["AstDec(deg)"],
+        }
+    )
 
-            # get the original index numbers of the observations that made it through
-            # append them to the final index
-            final_index = obs_final_night["original_index"].values
-            final_idx.append(final_index.tolist())
+    metric = NightlyLinkagesMetric(
+        linkage_min_obs=min_observations,  # min observations per tracklet
+        max_obs_separation=maximum_time,  # maximum time separation between subsequent observations in a tracklet [days]
+        min_linkage_nights=min_tracklets,  # minimum number of nights on which a tracklet must be detected
+        min_obs_angular_separation=minimum_separation,  # minimum angular separation between observations in a tracklet [arcseconds]
+    )
 
-    # flatten the final index
-    flat_idx = [x for xs in final_idx for x in xs]
+    findable, window_summary = metric.run(
+        difi_dataframe,
+        detection_window=tracklet_interval,  # number of nights in a rolling detection window
+        min_window_nights=min_tracklets,  # minimum size of a window as the rolling window is moved
+        by_object=True,  # split observations by object instead of by window [set to True if you want ignore_after_discovery=True]
+        discovery_probability=detection_efficiency,  # probability of discovery for any discovery chance
+        num_jobs=30,  # number of parallel jobs
```

Comment:
If num_jobs is set to None, it will automatically use all cores available on the system. But depending on how this is used in the simulator this might also need to be 1.

Reply:
@moeyensj if `num_jobs` is set to a value larger than the number of cores available, will `difi` just use the core that are available? Should this be made into a config parameter?